molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/OneClickImporterServiceImpl.java

```diff
@@ -162,8 +167,8 @@ private Column createColumnFromLine(String header, int columnIndex, List<String>
 		List<Object> dataValues = newLinkedList();
 		lines.forEach(line ->
 		{
-			String[] lineParts = line.split(CSV_SEPARATOR, -1);
-			dataValues.add(getPartValue(lineParts[columnIndex]));
+			String[] tokens = line.split(CSV_SEPARATOR + "(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)", -1);
```

Comment:
consider using named function and adding javadoc, then try to refactor using String class